olive/engine/engine.py

```diff
@@ -227,47 +263,64 @@ def run(
+        for accelerator_spec in self.accelerator_specs:
             # generate search space and intialize the passes for each hardware accelerator
-            self.setup_passes()
+            self.setup_passes(accelerator_spec)
 
             # hash the input model
             input_model_id = self._init_input_model(input_model)
-            self.footprints[i].record(model_id=input_model_id)
-
-            if evaluation_only:
-                prefix_output_name = f"{output_name}_{i}_" if output_name is not None else f"{i}_"
-                assert self.evaluator is not None, "Evaluation only is True but no evaluator provided"
-                results = self._evaluate_model(input_model, input_model_id, self.evaluator, i, verbose)
-                result_name = f"{prefix_output_name}metrics"
-                results_path = output_dir / f"{result_name}.json"
-                json.dump(results, open(results_path, "w"), indent=4)
-                outputs[i] = results
-            elif self.no_search:
-                output = self.run_no_search(
-                    input_model, input_model_id, i, packaging_config, verbose, output_dir, output_name
-                )
-                outputs[i] = output
-            else:
-                footprint = self.run_search(
-                    input_model, input_model_id, i, packaging_config, verbose, output_dir, output_name
-                )
-                outputs[i] = footprint
+            self.footprints[accelerator_spec].record(model_id=input_model_id)
+
+            try:
+                if evaluation_only:
+                    prefix_output_name = (
+                        f"{output_name}_{accelerator_spec}_" if output_name is not None else f"{accelerator_spec}_"
+                    )
+                    assert self.evaluator is not None, "Evaluation only is True but no evaluator provided"
+                    results = self._evaluate_model(
+                        input_model, input_model_id, self.evaluator, accelerator_spec, verbose
+                    )
+                    result_name = f"{prefix_output_name}metrics"
+                    results_path = output_dir / f"{result_name}.json"
+                    json.dump(results, open(results_path, "w"), indent=4)
+                    outputs[accelerator_spec] = results
+                elif self.no_search:
+                    output = self.run_no_search(
+                        input_model,
+                        input_model_id,
+                        accelerator_spec,
+                        packaging_config,
+                        verbose,
+                        output_dir,
+                        output_name,
+                    )
+                    outputs[accelerator_spec] = output
+                else:
+                    footprint = self.run_search(
+                        input_model,
+                        input_model_id,
+                        accelerator_spec,
+                        packaging_config,
```

Comment:
Since we have multiple footprints now, the packaging only has one zipfile name in the config. So current logic will override previous `accelerator_spec` with the same zipfile name and eventually only leave one zipfile with a single `accelerator_spec` case. I would suggest updating packaging logic to have multiple folders with different name for each `accelerator_spec`.

Reply:
could you please help provide some guidance on how to change the logic? I am not familiar with the package logic.

Reply:
@xiaoyu-work , please help take a look the package generation logic.